NSS-Day-Cohort-69/tabloid-one-more-div
# Md feature/35/edit a category

### **Purpose**
To Allow a user with the role admin to create a category

### **Files Changed**
- Created file, and added the class with the needed properties in new `CategoryUpdate.cs`
- Added the endpoint for Editing a Category in `CategoryController.cs`
- Added the fetch for editing a Category in `categoryManager.js`
- Changed file name `CategoryCreate.jsx to `CreateCategoryForm.jsx`
- Added the logic to the handleSubmit so that if categoryId is present then it runs the function that sends the updated category to the controller, otherwise it will run the function that creates a category in `CreateCategoryForm.jsx`
-  added routing for Categories when the user clicks create or edit the user is taken to the corresponding view in `ApplicationView.jsx`

### **To Test**

Fetch ,setup and run according to project README, then confirm the following

- Logged in Admin can click Edit on a category 
- Logged in Admin gets navigated to a form to Edit the category
- Logged in Admin clicks save 
- Logged in Admin gets navigated to the Category Management page and sees the updated category
